lib/src/analyzers/lint_analyzer/rules/rules_list/format_comment/visitor.dart

```diff
@@ -44,20 +44,31 @@ class _Visitor extends RecursiveAstVisitor<void> {
 
     var text = commentText.trim();
 
-    if (text.isEmpty ||
-        text.startsWith('ignore:') ||
-        text.startsWith('ignore_for_file:')) {
-      return;
-    } else {
-      text = text.trim();
-      final upperCase = text[0] == text[0].toUpperCase();
-      final lastSymbol = _punctuation.contains(text[text.length - 1]);
-      final hasEmptySpace = commentText[0] == ' ';
-      final incorrectFormat = !upperCase || !hasEmptySpace || !lastSymbol;
-      final single = commentToken.previous == null && commentToken.next == null;
+    final isIgnoreComment =
+        text.startsWith('ignore:') || text.startsWith('ignore_for_file:');
 
-      if (incorrectFormat && single) {
-        _comments.add(_CommentInfo(type, commentToken));
+    final regTemplateExp = RegExp(r"{@template [\w'-]+}");
+    final regMacroExp = RegExp(r"{@macro [\w'-]+}");
```

Comment:
I would make them file-private final variables. It can look like a micro-optimization, but for analyzer running on thousands comments that could matter.

E.g. this benchmark:

```dart
import 'package:benchmark/benchmark.dart';

void main() {
  benchmark('Create RegExp', () {
    List.generate(100000, (i) {
      final x = RegExp(r"{@template [\w'-]+}");
      return x.hasMatch('abc');
    });
  });

  benchmark('Reise RegExp', () {
    List.generate(100000, (i) {
      return _regex.hasMatch('abc');
    });
  });
}

final _regex = RegExp(r"{@template [\w'-]+}");
```

gives such a result:

```
❯ dart run benchmark --dir=.
 DONE  ././lib/example_benchmark.dart (5 s)
 ✓ Create RegExp (103 ms)
 ✓ Reise RegExp (22 ms)

Benchmark suites: 1 passed, 1 total
Benchmarks:       2 passed, 2 total
Time:             5 s
Ran all benchmark suites.
```

Reply:
Nice catch, actually!